tests/schema/tests.py

```diff
@@ -1673,3 +1673,20 @@ def test_alter_field_default_dropped(self):
             )
             if connection.features.can_introspect_default:
                 self.assertIsNone(field.default)
+
+    def test_add_textfield_unhashable_default(self):
+        # Create the table
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+        # Create a row
+        Author.objects.create(name='Anonymous1')
+        # Create a field that has an unhashable default
+        new_field = TextField(default={})
+        new_field.set_attributes_from_name("info")
+
+        def get_db_prep_save(default, connection):
+            return str(default)
+        new_field.get_db_prep_save = get_db_prep_save
+        # Fail if an exception is raised
+        with connection.schema_editor() as editor:
+            editor.add_field(Author, new_field)
```

Comment:
It's better to wrap this with try/except like below so when test fails we get a failure instead of error. 

```python
        # Fail if an exception is raised
        try:
            with connection.schema_editor() as editor:
                editor.add_field(Author, new_field)
        except TypeError:
            self.fail("This should not fail with TypeError")
```